core/templates/dev/head/domain/exploration/ReadOnlyExplorationBackendApiService.js

```diff
@@ -14,25 +14,35 @@
 
 /**
  * @fileoverview Service to retrieve read only information
- * about exploraions from the backend.
+ * about explorations from the backend.
  */
 
 oppia.factory('ReadOnlyExplorationBackendApiService', [
   '$http', '$q', 'EXPLORATION_DATA_URL_TEMPLATE',
-  'UrlInterpolationService',
+  'EXPLORATION_VERSION_DATA_URL_TEMPLATE', 'UrlInterpolationService',
   function($http, $q, EXPLORATION_DATA_URL_TEMPLATE,
-    UrlInterpolationService) {
+    EXPLORATION_VERSION_DATA_URL_TEMPLATE, UrlInterpolationService) {
     // Maps previously loaded explorations to their IDs.
     var _explorationCache = [];
 
     var _fetchExploration = function(
-        explorationId, successCallback, errorCallback) {
-      var explorationDataUrl = UrlInterpolationService.interpolateUrl(
+        explorationId, version, successCallback, errorCallback) {
+      var explorationDataUrl = '';
```

Comment:
Consider moving the if (version) logic to a helper method that returns one option or the other, that way we can directly initialize explorationDataUrl and are never in a state where it's an invalid value.